pilot/pkg/model/push_context.go

```diff
@@ -347,7 +347,11 @@ func (ps *PushContext) SetSidecarScope(proxy *Proxy) {
 
 // Services returns the list of services that are visible to a Proxy in a given config namespace
 func (ps *PushContext) Services(proxy *Proxy) []*Service {
-	if proxy != nil && proxy.SidecarScope != nil && proxy.Type == SidecarProxy {
+	// If proxy has a sidecar scope that is user supplied, then get the services from the sidecar scope
+	// sidecarScope.config is nil if there is no sidecar scope for the namespace
+	// TODO: This is a temporary gate until the sidecar implementation is stable. Once its stable, remove the
+	// config != nil check
+	if proxy != nil && proxy.SidecarScope != nil && proxy.SidecarScope.Config != nil && proxy.Type == SidecarProxy {
```

Comment:
Qq: what happens for Router/Ingress ? Different code path ?

Reply:
And BTW - we still need to implement ( in separate PR ) the Sidecar spec for gateways.

Reply:
gateways dont use sidecar scope. Gateway definition is sufficient to cover the cases.